internal/indexer/db/client/delegation.go

```diff
@@ -66,27 +95,26 @@ func (indexerdbclient *IndexerDatabase) GetDelegations(
 
 		orConditions := []bson.M{
 			{
-				"staker_btc_pk_hex":                       stakerPKHex,
 				"btc_delegation_created_bbn_block.height": bson.M{"$lt": decodedToken.StartHeight},
 			},
 			{
-				"staker_btc_pk_hex":                       stakerPKHex,
 				"btc_delegation_created_bbn_block.height": decodedToken.StartHeight,
 				"_id": bson.M{"$gt": decodedToken.StakingTxHashHex},
 			},
 		}
-		if stakerBabylonAddress != nil {
-			addr := *stakerBabylonAddress
-			orConditions[0]["staker_babylon_address"] = addr
-			orConditions[1]["staker_babylon_address"] = addr
+
+		for _, filter := range filters {
+			filter(orConditions[0])
+			filter(orConditions[1])
```

Comment:
this is my fav. part: now we won't break paginationKey even if we remove/add new filter for paginations in future